### PR TITLE
Analytics wizard

### PR DESCRIPTION
**Pull request by irlaec**

### CODE REVIEW

File: octoprint_mrbeam/analytics/analytics_handler.py [hunk 1]
```diff
@@ -147,13 +153,16 @@ def _days_passed(path_to_file):
 
 	def analytics_user_permission_change(self, analytics_enabled):
 		self._logger.info("analytics user permission change: analyticsEnabled=%s", analytics_enabled)
+		if self._analyticsOn is None:
```
Comment (andywerner) [inline octoprint_mrbeam/analytics/analytics_handler.py]: @Gallore Is this really the right thing here?
When the user agrees or disagrees for the first time, we start the uploader in any case?

File: octoprint_mrbeam/analytics/analytics_handler.py [hunk 2]
```diff
@@ -603,3 +612,39 @@ def _append_data_to_file(self, data):
...
+			self.process_analytics_files()
+
+		elif consent == 'disagree':
+			self.analytics_user_permission_change(False)
+			self.delete_analytics_files()
+
+	def delete_analytics_files(self):
+		self._logger.info("Deleting analytics files...")
+		folder = ak.ANALYTICS_FOLDER
+		for file in os.listdir(folder):
```
Comment (andywerner) [inline octoprint_mrbeam/analytics/analytics_handler.py]: is file a key-word in python? if yes, pls change

File: octoprint_mrbeam/analytics/analytics_handler.py [hunk 3]
```diff
@@ -603,3 +612,39 @@ def _append_data_to_file(self, data):
...
+				if os.path.isfile(file_path) and "analytics" in file:
+					os.unlink(file_path)
+					self._logger.info('File deleted: {file}'.format(file=file_path))
+			except Exception as e:
+				self._logger.error('Error when deleting file {file}: {error}'.format(file=file_path, error=e))
+
+	def process_analytics_files(self):
+		self._logger.info("Processing analytics files...")
+		folder = ak.ANALYTICS_FOLDER
+		for file in os.listdir(folder):
```
Comment (andywerner) [inline octoprint_mrbeam/analytics/analytics_handler.py]: file key word

File: octoprint_mrbeam/static/js/wizard_analytics.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
+$(function() {
+    function WizardAnalyticsViewModel(parameters) {
+         var self = this;
+
+         self.analyticsInitialConsent = ko.observable(null);
+
+         self.saveAnalyticsConsent = function(){
+		    let data = {analyticsInitialConsent: self.analyticsInitialConsent}
+		    console.log("######################### save analytics consent: " + data)
```
Comment (andywerner) [inline octoprint_mrbeam/static/js/wizard_analytics.js]: pls remove debug output

File: octoprint_mrbeam/static/js/wizard_analytics.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
+$(function() {
+    function WizardAnalyticsViewModel(parameters) {
+         var self = this;
+
+         self.analyticsInitialConsent = ko.observable(null);
+
+         self.saveAnalyticsConsent = function(){
+		    let data = {analyticsInitialConsent: self.analyticsInitialConsent}
```
Comment (andywerner) [inline octoprint_mrbeam/static/js/wizard_analytics.js]: pls check why there are no parentheses at the end of "self.analyticsInitialConsent"
as a ko.observable you ususally call it per "self.analyticsInitialConsent()"

File: octoprint_mrbeam/static/js/wizard_analytics.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+		    console.log("######################### save analytics consent: " + data)
+
+		    OctoPrint.simpleApiCommand("mrbeam", "analytics_init", data)
+                .done(function(response){
+					console.log("simpleApiCall response: ", response);
```
Comment (andywerner) [inline octoprint_mrbeam/static/js/wizard_analytics.js]: pls add some more descriptive log output

File: octoprint_mrbeam/static/js/wizard_analytics.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+         // }
+
+         self.onAfterWizardTabChange = function(current) {
+             if (current === "wizard_plugin_corewizard_analytics_link") {
+                 if (self.analyticsInitialConsent() == null) {
+                     $("#wizard_dialog div.modal-footer button.button-finish").prop("disabled", true);
+                     $("#wizard_dialog div.modal-footer button.button-finish").removeAttr("onclick");
+                 }
+
+                 // $("#wizard_dialog div.modal-footer button.button-finish").attr("data-bind", "enable: responded(), click: agree");
```
Comment (andywerner) [inline octoprint_mrbeam/static/js/wizard_analytics.js]: pls delete

File: octoprint_mrbeam/static/js/wizard_analytics.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+					new PNotify({
+                        title: "Error while saving settings!",
+                        text: "Unable to save your analytics state at the moment.<br/>Check connection to Mr Beam II and try again.",
+                        type: "error",
+                        hide: true
+                    });
+				});
+         }
+
+         // self.onBeforeWizardTabChange = function(next, current) {
```
Comment (andywerner) [inline octoprint_mrbeam/static/js/wizard_analytics.js]: delete

File: octoprint_mrbeam/static/js/wizard_analytics.js [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+            return elem ? (!!ko.dataFor(elem)) : false;
+         };
+
+         self.enableLetsGoButton = function() {
+             $("#wizard_dialog div.modal-footer button.button-finish").prop("disabled", false);
+             $("#wizard_dialog div.modal-footer button.button-finish").click(function () {
+                 let consent = self.analyticsInitialConsent()
+                 let data = {analyticsInitialConsent: consent}
+                 console.log("######################### save analytics consent: " + consent)
+                 OctoPrint.simpleApiCommand("mrbeam", "analytics_init", data)
```
Comment (andywerner) [inline octoprint_mrbeam/static/js/wizard_analytics.js:25]: check why this code part is here two times. have it only once.

File: octoprint_mrbeam/static/js/wizard_analytics.js [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+
+             }
+         }
+
+         self.is_bound = function(){
+            var elem = document.getElementById(DOM_ELEMENT_TO_BIND_TO);
+            return elem ? (!!ko.dataFor(elem)) : false;
+         };
+
+         self.enableLetsGoButton = function() {
```
Comment (andywerner) [inline octoprint_mrbeam/static/js/wizard_analytics.js]: naming: the name does not indicate that the function also uploads data.

**Review by andywerner — CHANGES_REQUESTED** [review]

File: octoprint_mrbeam/analytics/analytics_handler.py (hunk 1 above)
Comment (irlaec) [reply octoprint_mrbeam/analytics/analytics_handler.py]: Changed to only activate upload when the analytics are accepted the first time

File: octoprint_mrbeam/static/js/wizard_analytics.js (hunk 5 above)
Comment (irlaec) [reply octoprint_mrbeam/static/js/wizard_analytics.js]: SaveAnalyticsConsent was not used, I have deleted it.

File: octoprint_mrbeam/static/js/wizard_analytics.js (hunk 8 above)
Comment (irlaec) [reply octoprint_mrbeam/static/js/wizard_analytics.js]: Done

File: octoprint_mrbeam/static/js/wizard_analytics.js (hunk 7 above)
Comment (irlaec) [reply octoprint_mrbeam/static/js/wizard_analytics.js]: Done

File: octoprint_mrbeam/static/js/wizard_analytics.js (hunk 9 above)
Comment (irlaec) [reply octoprint_mrbeam/static/js/wizard_analytics.js:25]: The other one was unused, now it is deleted

File: octoprint_mrbeam/analytics/analytics_handler.py (hunk 2 above)
Comment (irlaec) [reply octoprint_mrbeam/analytics/analytics_handler.py]: Changed

File: octoprint_mrbeam/analytics/analytics_handler.py (hunk 3 above)
Comment (irlaec) [reply octoprint_mrbeam/analytics/analytics_handler.py]: Changed

File: octoprint_mrbeam/static/js/wizard_analytics.js (hunk 10 above)
Comment (irlaec) [reply octoprint_mrbeam/static/js/wizard_analytics.js]: Renamed

**Review by andywerner — APPROVED** [review]

**Review by andywerner — APPROVED** [review]

**Pull request merged**